### PR TITLE
Add new allow_constraints_for_unverifiable_logs config

### PR DESCRIPTION
**Pull request by dosaboy**
Disables constraints for files that contain unverifiable lines and adds a config option to re-enable them. This is until we have a way to apply constraints for files containing lines that dont have timestamps without it being really slow for large files.

Resolves: #498

### CODE REVIEW

File: hotsos/core/config.py [hunk 1]
```diff
@@ -112,6 +112,25 @@ def __init__(self):
                                         'log files, this is used to limit the '
                                         'logrotate history in days.'),
                            default_value=7))
+        self.add(ConfigOpt(name='allow_constraints_for_unverifiable_logs',
+                           description=('Search constraints use a binary '
+                                        'search that sometimes needs to '
+                                        'search backwards to find a last '
+                                        'known good line i.e. in log files '
+                                        "that contain lines that don't start "
```
Comment (brianphaley) [inline hotsos/core/config.py:120]: My nitpicking eyes were drawn to the "" here :-p

**Review by brianphaley — APPROVED** [review]

File: hotsos/core/config.py [hunk 2]
```diff
@@ -112,6 +112,25 @@ def __init__(self):
...
+                                        'with a timestamp we treat those as '
+                                        'unverifiable and so have to find the '
+                                        'most recent verifiable line which '
+                                        'can be ahead or behind the current '
+                                        'position. Seeking backwards seems to '
+                                        'force a SEEK_SET in the kernel which '
```
Comment (pponnuvel) [inline hotsos/core/config.py:126]: I understand the issue - but I doubt it's the *kernel* setting SEEK_SET (I suspect it's somewhere in the Python lib).

**Review by pponnuvel — APPROVED** [review]
LGTM

File: hotsos/core/config.py (hunk 1 above)
Comment (dosaboy) [reply hotsos/core/config.py:120]: thank brian, yeah that's cause of the ' in the string

File: hotsos/core/config.py (hunk 2 above)
Comment (dosaboy) [reply hotsos/core/config.py:126]: yeah I can reword this as its not entirely correct, basically what I see is that if i am at pos 100002 and want to seek(SEEK_SET, 100001) it will seek from the start of the file which may be intentional but if i do seek(SEEK_CUR, -1) it also forces to SEEK_SET and starts from 0 (which i can see in strace).

File: hotsos/core/config.py (hunk 1 above)
Comment (brianphaley) [reply hotsos/core/config.py:120]: Oh yeah, didn't (or did not) even notice :)

File: hotsos/core/config.py (hunk 2 above)
Comment (dosaboy) [reply hotsos/core/config.py]: ive updated the description fwiw
